src/datachain/lib/dc.py

```diff
@@ -904,6 +904,23 @@ def agg(
             )
             chain.save("new_dataset")
             ```
+
+            ```py
+            # It automatically resolves which columns to pass to the function
+            # by looking at the function signature.
+            def agg_sum(file: list[File],
+                        amount: list[float]) -> Iterator[tuple[File, float]]:
+                yield file[0], sum(amount)
+
+            chain = chain.agg(
+                agg_sum,
+                output={"file": File, "total": float},
+                # Alternative syntax is to use `C`
```

Comment:
I would maybe merge these 2 sentences into one... "Alternative syntax is to use `C` which is useful for nested columns ..."